server/src/services/asset.service.ts

```diff
@@ -97,6 +97,8 @@ export class AssetService extends BaseService {
 
     const asset = await this.assetRepository.update({ id, ...rest });
 
+    await this.eventRepository.emit('asset.updated', { assetId: id, userId: auth.user.id });
```

Comment:
Why don't you just call this with the asset, since you already have it?

Reply:
I thought about it - but aren't these events bull or websocket events? Is there any problem with sending a lot of information in the event payload itself?

Reply:
`.emit` triggers `@OnEvent` directly/immediately, and even awaits it. It just triggers a method in another place in the code. It's in the same process and really only exists so that notification logic doesn't have to exist in all these other services. 

`@OnJob` or whatever exists for jobs which so come from bull/redis.